README.md

```diff
@@ -1,24 +1,100 @@
-let _scope = metrics.time("timed_delay"); // you can time additional scopes
+let mut metrics = metrics_factory.record_scope("demo"); // By default, includes a "demo_totaltime" histogram measurement, capturing the time it took to complete the unit of work
+{
+    let _scope = metrics.time("timed_delay"); // you can time additional scopes
+    my_timed_delay();
+}
 metrics.measurement("ran", 1); // measurements can be plain numbers; when preaggregated they are StatisticSets (min/max/sum/count)
+metrics.sum("runs_count", 1); // If you do not need a StatisticSet but rather a simple counter,
+                              // sum will produce a simple Gauge
 metrics.dimension("mod", i % 8); // you can add dimensions to a Metrics whenever you want. All measurements in this Metrics record are dimensioned by this value.
 metrics.distribution("some_continuous_value", instantaneous_network_bandwidth); // histograms are aggregated sparsely, and truncated to 2 significant figures (base 10).
 ```
 
-# Development
+Once the `metrics` object is dropped from scope, it will export metrics to your desired ingest. If you need to publish metrics at some immediate point, you can manually `drop()` the object.
+
+### Record scope without measuring time
+
+Not every unit of work needs a measurement of how long it took to complete. Simply create a metrics object by calling `record_scope_with_behavior` and passing in your desired behavior.
+
+For example:
+
+```rust
+let mut metrics = metrics_factory.record_scope_with_behavior(
+    "demo",
+    MetricsBehavior::SuppressTotalTime,
+);
+```
+
+### Create a dimension with a default value
+
+In a distributed system, a unit of work may not fully complete and the `metrics` object will fall out of scope, causing any existing dimensions or measurements to be exported. This can lead to confusing behavior if a dimension was expected, but not present.
+
+Using a `guarded_dimension` can provide a default value and export it should the object be dropped before expected:
+
+```rust
+let mut metrics = metrics_factory.record_scope("demo");
+let result_dimension = metrics.guarded_dimension("my_api_result", "dropped_early");
+
+// Perform work...
+let result = serve_api_request(request);
+
+match result {
+    Ok(_) => {
+        // Explicitly set the guarded dimension and overwrite the default value
+        result_dimension.set("ok")
+    },
+    Err(_e) => {
+        // Explicitly set the guarded dimension and overwrite the default value
+        result_dimension.set("error");
+        metrics.sum("errors", 1_u64)
+    }
+}
+```
+
+### Available metrics functions and when to use them
+
+Latest docs can always be found on [docs.rs](https://docs.rs/goodmetrics/latest/goodmetrics/).
+
+A basic knowledge of [OpenTelemetry metric types](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#metric-points) is expected, but a shortened guide can be summarized below:
```

Comment:
I think if you want to put this in here it should go in a separate heading below this, `### Opentelemetry translation` or something like that. The translation per-type is kind of interesting, and probably deserves some explanation!

Reply:
Upon further reflection. we can always omit this

Reply:
https://github.com/kvc0/goodmetrics_rs/pull/19/commits/aadb97db80102ea54997e25e0db4f983fd1ec7b0